doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,432 @@
+    |                                   | `zcbor_tstr_encode_ptr`           |
+    |                                   | `zcbor_tstr_put_lit`              |
+    +-----------------------------------+-----------------------------------+
+    | `cbor_encode_text_stringz`        | `zcbor_tstr_put_term`             |
+    +-----------------------------------+-----------------------------------+
+    | `cbor_encode_byte_string`         | `zcbor_bstr_encode`               |
+    |                                   | `zcbor_bstr_encode_ptr`           |
+    +-----------------------------------+-----------------------------------+
+    | `cbor_encode_null`                | `zcbor_nil_put`                   |
+    +-----------------------------------+-----------------------------------+
+    | `cbor_encode_undefined`           | `zcbor_undefined_put`             |
+    +-----------------------------------+-----------------------------------+
+
+There is no direct replacement for the `cbor_value_is_null` and `cbor_value_is_undefined` functions in zcbor.
+To check whether such value is where expected, you can utilize the following code:
+
+.. code-block:: c
+
+   bool ok;
+   ...
+   /* Make sure that not in error state first */
+   if (zs->constant_state->error != 0) {
+       /* Already in error */
+       ...
+   }
+   ok = zcbor_nil_decode(zs, NULL);
+   if (!ok && zs->constant_state->error == ZCBOR_ERR_WRONG_TYPE) {
+       /* Process non NULL element */
+       zcbor_pop_error(zs);
+       ...
+   } else if (!ok) {
+       /* Other error */
+       ...
+   }
+   /* Successfully decoded NULL element in CBOR stream */
+   ...
+
+.. _migration_tinycbor_to_zcbor_mcumgr:
+
+MCUmgr commands and groups migration
+====================================
+
+In contrast to :ref:`migration_tinycbor_to_zcbor_general`, except for tuning some Kconfig configuration constants, there is no need for zcbor setup as the MCUmgr library takes case of the task.
+Most of the task which consists of function replacement is actually covered by the :ref:`migration_tinycbor_to_zcbor_general`, with exception to usage of `cborattr` which is covered by paragraph :ref:`migration_tinycbor_to_zcbor_mcumgr_cborattr`.
+
+The other important issue is decoding/encoding context access, which is described in :ref:`migration_tinycbor_to_zcbor_mcumgr_ctxt`.
+
+.. note::
+
+   The encoding context given to a command processing handler function has the top level map already created.
+   It will have the map closed on successful return from the handler, so handlers need not create the top level map.
+   In contrary to the above, the decoding context is not yet opened and handlers are responsible for opening the top level map before decoding of a command.
+
+.. _migration_tinycbor_to_zcbor_mcumgr_ctxt:
+
+Decoding and encoding context
+-----------------------------
+
+To command the processing of functions in TinyCBOR and zcbor implementation, the MCUmgr provides the following:
```

Comment:
```suggestion
To allow you to have control over the processing of functions in TinyCBOR and zcbor implementation, MCUmgr provides the following resources:
```
@de-nordic , please check the correctness of this edit.

Reply:
Working was clumsy: "command processing function" , should be a function (handler), so probably more correct would be: "To a handler function for a SMP command, in TinyCbor and zcbor implementation, ..."